### ISSUE TITLE
Calculate the Daily average for Agile Tarif

### ISSUE THREAD

**Issue by LucidityCrash** [BottlecapDave/HomeAssistant-OctopusEnergy#214]
I've been using OctopusAgile but since it isn't maintained anymore I've been looking at this.  One feature I'm missing is the Average for the day, which I've used to work out if I should force charge my solar batteries during the cheapest 3hr slot but only if the average for that slot is less than the average for the whole day. 

I could probably use a template and calculate it myself but it might be easier to do in the code.

**Comment by BottlecapDave** [comment]
I'll add it to the backlog. In the meantime, you can calculate the daily average using a template sensor with the current setup

```
{{ (state_attr('sensor.octopus_energy_electricity_{{METER_SERIAL_NUMBER}}_{{MPAN_NUMBER}}_current_rate','rates') | sum(attribute='rate')) / (state_attr('sensor.octopus_energy_electricity_{{METER_SERIAL_NUMBER}}_{{MPAN_NUMBER}}_current_rate','rates') | count) / 100 }}
```

**Comment by LucidityCrash** [comment]
Cool, thanks for the template.

Something of interest though is my sensors are not
octopus_energy_electricity_ or gas  etc but just electricity_ or gas_ might
have been an unintended consequence of renaming you did a few releases back
( I only installed a few days ago )

**Comment by BottlecapDave** [comment]
The ids of the entities should still start with `octopus_energy`. It should just be the names of the entities that don't include octopus energy.

**Comment by LucidityCrash** [comment]
Not for me I'm afraid :
![image](https://user-images.githubusercontent.com/49124523/231860494-85b909de-8641-45d5-903a-1e42d37ea082.png)

**Comment by BottlecapDave** [comment]
Damn, you're right. That is unfortunate. I'll try to rectify that in a future release.

**Comment by BottlecapDave** [comment]
Hello. Just to say

1. The naming of the entities have been updated to reflect the documentation
2. Attributes representing min/max/average are now available on the electricity rate sensors

These are available in the latest release (v7.1.0)
